AlexHamn/bookstore
# Create and remove books

### Changes in this branch:

- Configure the Redux Store (`/src/redux/configureStore.js`) and import the necessary methods from Redux:
    - Import `createStore`
    - Import `combineReducers` - We will only write a reducer for 'books' in this step, but in the future, we might want to add 'categories' or other reducers as well. 
    - Import `applyMiddleware` - To use some handy Redux middleware, like *redux logger*.
    - Import `booksReducer`, which is written in the `/src/redux/books` directory.
   
- Write book's actions and reducer (in `/src/redux/books/books.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define *action types*:
    ```javascript
    // books.js

    const ADD_BOOK = 'bookStore/books/ADD_BOOK';
    const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
    ```
    - Set the initial state to be an empty array of books:
    ```javascript
    // books.js
    ...
    const initialState = [];
    ```
    - Export *Action Creators* (functions, which return *actions*):
    ```javascript
    // books.js
    ...
    export const addBook = payload => ({
        type: ADD_BOOK,
        payload
    })
    export const removeBook = ...
    ```
    - Write reducer and export it as default.

- In the React component responsible for adding new books, set the data inputs in the local React state (set *title* and *author*  - NOTE: *categories* and *comments* are NOT part of this step). Once the new book object is ready to be submitted to the Redux store, dispatch a corresponding action.

- In the React component responsible for removing books - implement the event by dispatching a corresponding action.
- Use the *redux-logger* to check if the application is working correctly. Upon every action dispatch, the logger will print:
    - **prev state** - the state of the whole Redux store before the current action was dispatched.
    - **action** - the action object that is being dispatched.
    - **next state** - the state of the Redux store after the current action was dispatched.
